### DISCUSSION TITLE
Installation of next 12

### DISCUSSION THREAD

**Discussion by UmerGillani36** [vercel/next.js#50957] [Help] [answered]
### Summary

I want to create next app 12 with all the features of next 12 instead of latest 13 version, how can create and use ?

**Comment by icyJoseph** [comment]
Well, for full disclosure, `Next@13 = appDir + Next@12 pages dir + new next/link + new next/image + a few other bits`, perhaps the fonts package should be counted in too. 

I wonder if all you want is pages dir, because then you can go ahead with Next 13 no problem.

You can run:

```console
yarn create next-app --no-app --js your-project-name
```

Or with `npx`, and that'll get you:

```console
.
├── README.md
├── jsconfig.json
├── next.config.js
├── package.json
├── pages
│   ├── _app.js
│   ├── _document.js
│   ├── api
│   │   └── hello.js
│   └── index.js
├── public
│   ├── favicon.ico
│   ├── next.svg
│   └── vercel.svg
├── styles
│   ├── Home.module.css
│   └── globals.css
└── yarn.lock
```

If that's still not enough, you can then, do `yarn add next@12.1.3` for example, and that'll install that next version. Then you, most likely have to remove all `next/font/google` references from the starter code, which is just 1, and then you are good to go! :) 

## Alternatively

0. `mkdir my-project-name && cd my-project-name`
1. `yarn init -y`
2. `yarn add next@12.1.3 react react-dom`
3. Create a `pages/index.js` file
4. `npx next dev`

Just like that 😄

**Reply by UmerGillani36** [reply]
can i do this with , npx create-react-app@latest and when the app is created then i run the npm i next@12

**Reply by UmerGillani36** [reply]
and use the pages instead of app dir

**Reply by icyJoseph** [reply, marked as answer]
Yeah, exactly what I described:

```console
yarn create next-app --no-app your-project-name
```

The `--no-app` flag opts out from app dir. Then you can just install next@12, and it should be good to go.
